src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -148,13 +148,14 @@ public async Task<SubscriptionState> AssertSubscriptionConnectionDetails(long id
                     var databaseTopologyAvailabilityExplanation = new Dictionary<string, string>();
 
                     string generalState = string.Empty;
-                    if (_serverStore.Engine.CurrentState == RachisState.Candidate || _serverStore.Engine.CurrentState == RachisState.Passive)
+                    RachisState currentState = _serverStore.Engine.CurrentState;
+                    if (currentState == RachisState.Candidate || _serverStore.Engine.CurrentState == RachisState.Passive)
```

Comment:
You can use local variable in 2nd check also

Reply:
sure, missed it